### ISSUE TITLE
Issue: Slug for root `index.md` file generates wrong preview URL

### ISSUE THREAD

**Issue by estruyf** [estruyf/vscode-front-matter#806]
Hi Elio, 

looks like the issue is still there. I have installed latest Front Matter CMS `v10.1.0`.

```jsonc
// frontmatter.json
{
  "$schema": "https://frontmatter.codes/frontmatter.schema.json",
  "frontMatter.framework.id": "astro",
  "frontMatter.preview.host": "http://localhost:4321",
  "frontMatter.content.publicFolder": {
    "path": "src/assets",
    "relative": true
  },
  "frontMatter.content.pageFolders": [{
      "title": "Docs",
      "path": "[[workspace]]/src/content/docs/",
      "previewPath": "{{pathToken.relPath}}"
    }
  ]
}
```

```js
$ tree content
src
└── content
    └── docs
        └── index.md // http://localhost:4321/docs -> expected http://localhost:4321/
        └── test.md // http://localhost:4321/test
            └── nested
                └── index.md // http://localhost:4321/nested // This works
```

If I can help out with some additional context, feel free to ask.
Have a nice day 👋

_Originally posted by @theinfinit in https://github.com/estruyf/vscode-front-matter/issues/603#issuecomment-2117437985_

**Comment by estruyf** [comment]
@theinfinit I included a fix in the latest beta, would you be able to verify it?

**Comment by theinfinit** [comment]
Oh, that's great, thank you. Works like a charm! Paths, both in the root and nested, are correct. 

In the root it is displayed with a dot `http://localhost:4321/.`, but works just fine.

Cheers 👋

**Comment by estruyf** [comment]
@theinfinit, the `previewPath` property comes before the `slug.` That is why it does not matter if you add a slash or not after `{{pathToken.relPath}}`.

You can add the `"frontMatter.taxonomy.slugSuffix": "/",` setting to your config to add a trailing slash at the end.

Thinking of it, it would be a good option to enhance the settings to have a `previewTrailingSlash` option.

**Comment by estruyf** [comment]
> Mentioning "frontMatter.taxonomy.slugSuffix": "/", setting [in the "Site Preview" documentation](https://frontmatter.codes/docs/site-preview#configuration), could probably help with discoverability. 

Good feedback.

> Or, as an alternative idea, something like this might be quite intuitive too: "previewPath": "{{pathToken.relPath}}/{{slug}}/"

The preview path and the slug are intentionally separated, as the slug can depend on other properties and can be defined with the `frontMatter.taxonomy.slugTemplate` setting or `slugTemplate` property on the content type.

> And it might serve as a good groundwork for i18n improvements. Something like this would be beneficial: "previewPath": "{{localePath}}/{{pathToken.relPath}}".

`{{locale}}` should become its own placeholder which can be used in the preview path as well. Good feedback. I'll create a new issue for it to track it.

**Comment by theinfinit** [comment]
> The preview path and the slug are intentionally separated, as the slug can depend on other properties and can be defined with the `frontMatter.taxonomy.slugTemplate` setting or `slugTemplate` property on the content type.

Ah, I see. Then, all I can ask for is the improvement of the ["Site Preview" documentation](https://frontmatter.codes/docs/site-preview#configuration). Pointing to the slug related configuration for the "trailing slash" case, and the ["Slug" guide](https://frontmatter.codes/docs/content-creation/slug) in general, should avoid the confusion.

> `{{locale}}` should become its own placeholder which can be used in the preview path as well. Good feedback. I'll create a new issue for it to track it.

That's fantastic! I'm happy to hear that 😆

**Comment by estruyf** [comment]
I've been testing it a bit more, and adding the `trailingslash` setting/option will make it easier than using the `frontMatter.taxonomy.slugTemplate`.

In the upcoming beta, you can use the following:

### Globally

```json
"frontMatter.preview.trailingSlash": true
```

### Page folder level

```json
frontMatter.content.pageFolders": [
  {
    "title": "Root",
    "path": "[[workspace]]/src/content/docs/blog",
    "previewPath": "/blog",
    "trailingSlash": true
  }
]
```

### Content type level

```json
"frontMatter.taxonomy.contentTypes": [
  {
    "name": "simple",
    "pageBundle": false,
    "previewPath": null,
    "trailingSlash": true,
    "fields": []
  }
]
```

## To document

- [ ] Add global documentation -> `https://frontmatter.codes/docs/settings/overview#frontmatter.preview.trailingslash`
- [ ] Add page folder property
- [ ] Add content type property

**Comment by theinfinit** [comment]
I just tested latest beta `v10.2.9236459`, and you did a really good job. Everything works as expected in my environment on all 3 levels (globally, page folder and content type level). 

For the record, 
- I performed all tests with `previewPath` config containing trailing slash,
- for both root and nested paths,
- and for, `index.md` and `named-pages.md`

```
"frontMatter.content.pageFolders": [
  {
    "title": "Root",
    "path": "[[workspace]]/src/content/docs/",
    "previewPath": "{{pathToken.relPath}}/",
    "trailingSlash": true
  },
],
```

For me, the issue is resolved. Thank you!
